mono/utils/mono-compiler.h

```diff
@@ -113,5 +113,16 @@ typedef SSIZE_T ssize_t;
 #define MONO_GNUC_VERSION (__GNUC__ * 10000 + __GNUC_MINOR__ * 100 + __GNUC_PATCHLEVEL__)
 #endif
 
+/* Used to tell clang's ThreadSanitizer to not report data races that occur within a certain function */
+#if defined(__has_feature)
```

Comment:
I would suggest to leave it as it is for this PR but definitely think about expanding the macro, once more checkers are added that rely on `__has_feature`.